exporter/otlpexporter/exporter.go

```diff
@@ -97,19 +97,24 @@ type grpcSender struct {
 }
 
 func newGrpcSender(config *Config) (sender, error) {
-	gs := &grpcSender{}
 	dialOpts, err := config.GRPCClientSettings.ToDialOptions()
 	if err != nil {
 		return nil, err
 	}
 
-	if gs.grpcClientConn, err = grpc.Dial(config.GRPCClientSettings.Endpoint, dialOpts...); err != nil {
+	var clientConn *grpc.ClientConn
+	if clientConn, err = grpc.Dial(config.GRPCClientSettings.Endpoint, dialOpts...); err != nil {
 		return nil, err
 	}
-	gs.traceExporter = otlptracecol.NewTraceServiceClient(gs.grpcClientConn)
-	gs.metricExporter = otlpmetriccol.NewMetricsServiceClient(gs.grpcClientConn)
-	gs.logExporter = otlplogcol.NewLogServiceClient(gs.grpcClientConn)
-	gs.metadata = metadata.New(config.GRPCClientSettings.Headers)
+
+	gs := &grpcSender{
```

Comment:
nit: You don't need the `gs` variable anymore, can just return inline.

Reply:
I had that before, but I kind of don't like that the error nil is most invisible.

Reply:
Also a decent compiler should inline that so not worried that much